pkg/mbset/mbset.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package mbset
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"sync"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/tetragon/pkg/api/processapi"
+	"github.com/cilium/tetragon/pkg/bpf"
+)
+
+const (
+	MapName   = "tg_mbset_map"
+	InvalidID = ^uint32(0)
+	MaxIDs    = 64 // this value should correspond to the number of bits we can fit in mbset_t
+)
+
+type bitSet = uint64
+
+func openMap() (*ebpf.Map, error) {
+	fname := filepath.Join(bpf.MapPrefixPath(), MapName)
+	ret, err := ebpf.LoadPinnedMap(fname, &ebpf.LoadPinOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", fname, err)
+	}
+	return ret, nil
+}
+
+type State struct {
+	mu       sync.Mutex
+	nextID   uint32
+	mbsetMap *ebpf.Map
+}
+
+func newState() (*State, error) {
+	m, err := openMap()
+	if err != nil {
+		return nil, fmt.Errorf("failed to open map: %w", err)
+	}
+	return &State{
+		mbsetMap: m,
+	}, nil
+}
+
+func (s *State) Close() error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.mbsetMap.Close()
+}
```

Comment:
This does not seem to be used in this PR. Maybe remove that? As far as I can understand, the plan is to create the state once and have it throughout the execution of the agent. So there are no plans to close that.

Reply:
I'd like to leave it so that if we ever use it directly (e.g,. for testing) we would now that we need to release resources for it.

Generally, when using global state I always try to implement everything as if we did not have global state, and then add a global variable on top of that. My intention is to i) make it easy to test, and ii) allow us to migrate to something that does not use global state, like Cilium's hive (https://github.com/cilium/hive). We are far from it at this point, but one can dream.